PhysicsTools/IsolationAlgos/plugins/CandIsolatorFromDeposits.cc

```diff
@@ -108,15 +102,12 @@ CandIsolatorFromDeposits::SingleDeposit::SingleDeposit(const edm::ParameterSet &
       evdepVetos_.push_back(evdep);
   }
   std::string weight = iConfig.getParameter<std::string>("weight");
-  if (isNumber(weight)) {
-    //std::cout << "Weight is a simple number, " << toNumber(weight) << std::endl;
-    weight_ = toNumber(weight);
+  try {
```

Comment:
If we really go with the exception approach (which we generally disfavor), please add
```suggestion
  CMS_SA_ALLOW try {
```
to avoid static analyzer complaints.